cli/azd/pkg/project/internal/project_utils.go

```diff
@@ -34,3 +37,25 @@ func CreateDeployableZip(appName string, path string) (string, error) {
 
 	return zipFile.Name(), nil
 }
+
+const c_gitIgnore string = ".gitignore"
+
+// CreateSkipPatternsFromGitIgnore inspect root project path and a `servicePath`
+// to see if there is a .gitignore file. It then combine both files in a single list
+// of exclusions.
+func CreateSkipPatternsFromGitIgnore(servicePath string) ([]gitignore.GitIgnore, error) {
+	// azdContext will provide the azd-project root path
+	azdContext, err := azdcontext.NewAzdContext()
+	if err != nil {
+		return nil, err
+	}
+
+	rootPath := azdContext.ProjectDirectory()
+	// If .gitignore can't be open is fine, it could be missing
+	var allPatterns []gitignore.GitIgnore
+	rootPatterns, _ := gitignore.NewFromFile(filepath.Join(rootPath, c_gitIgnore))
```

Comment:
If `NewFromFile` returns an error, then `rootPaterns` will be `nil`, which will call issues later.  We need to guard the appends to only append when `NewFromFile` returns a `nil` error.

Reply:
Fixed to only add if err == nil. Thank you